sos/report/plugins/__init__.py

```diff
@@ -999,6 +1005,18 @@ def is_installed(self, package_name):
             len(self.policy.package_manager.all_pkgs_by_name(package_name)) > 0
         )
 
+    def is_snap_installed(self, package_name):
+        """Is the snap package $package_name installed?
+
+        :param package_name:    The name of the package to check
+        :type package_name:     ``str``
+
+        :returns: ``True`` if the snap package is installed, else ``False``
+        :rtype: ``bool``
+        """
+        pkg = self.policy.package_manager.pkg_by_name(package_name)
+        return pkg and pkg['pkg_manager'] == 'snap'
```

Comment:
`pkg_by_name()` returns `None` for any named package that is not known to the package manager, the `return` does not guarantee a `bool` to be returned:

```python
>>> from sos.policies.package_managers.rpm import RpmPackageManage
>>> pm = RpmPackageManager()

>>> def is_snap_installed(package_name):
...     pkg = pm.pkg_by_name(package_name)
...     return pkg and pkg['pkg_manager'] == 'snap'
... 
>>> is_snap_installed('kernel')
False
>>> is_snap_installed('notapkg')
>>> 
>>> out = is_snap_installed('notapkg')
>>> type(out)
<class 'NoneType'>
>>> 
```


This doesn't affect the `any()` call in practice, but if we end up using this elsewhere, it could throw devels off.

Reply:
okie, I'll get this refactored